modyn/supervisor/internal/triggers/cost/avoidablemissclassification_costtrigger.py

```diff
@@ -38,13 +49,39 @@ def inform_new_model(
     ) -> None:
         """Update the cost and performance trackers with the new model
         metadata."""
+
+        # Call CostTrigger's inform_new_model method to update the cost tracker
         super().inform_new_model(most_recent_model_id, number_samples, training_time)
 
+        # Call the internal PerformanceTriggerMixin's inform_new_model method to update the performance tracker
+        self._inform_new_model(most_recent_model_id, self._last_detection_interval)
+
     # ---------------------------------------------------------------------------------------------------------------- #
     #                                                     INTERNAL                                                     #
     # ---------------------------------------------------------------------------------------------------------------- #
 
     @override
     def _compute_regret_metric(self, batch: list[tuple[int, int]], batch_start: int, batch_duration: int) -> float:
         """Compute the regret metric for the current state of the trigger."""
-        raise NotImplementedError()
+
+        self.data_density.inform_data(batch)
+        num_samples, num_misclassifications, evaluation_scores = self._run_evaluation(interval_data=batch)
+
+        self.performance_tracker.inform_evaluation(
+            num_samples=num_samples,
+            num_misclassifications=num_misclassifications,
+            evaluation_scores=evaluation_scores,
+        )
+
+        estimated_new_avoidable_misclassifications, _ = (
+            self.misclassification_estimator.estimate_avoidable_misclassifications(
+                update_interval_samples=self.config.evaluation_interval_data_points,
+                data_density=self.data_density,
+                performance_tracker=self.performance_tracker,
+                method=self.config.forecasting_method,
+            )
+        )
+
+        return self._incorporation_latency_tracker.add_latency(
```

Comment:
I think this might be related to the doc PR but I still don't have a good understanding how how the num of avoidable misclassifications functions as a regret. for the incorporation latencies it's clear to me that each sample's regret grows with each timestep it's not incorporated. how come that the num avoided samples's regret also grows over time? this - in my understanding - is a static value and does not get more severe over time (unlike the incorporation latency). Can you elaborate on this either in the docstring or in the docs PR potentially? thank you!

Reply:
I'll add a comment in both places. Let's continue [here](https://github.com/eth-easl/modyn/pull/600#discussion_r1735130216)